src/test/setup.ts

```diff
@@ -150,10 +151,21 @@ const apiResponses: ApiResponses = {
     },
   },
 };
-export const restHandlers = [
-  rest.all('https://aha.rainbow.me/', (req, res, ctx) => {
-    const address = req.url.searchParams.get('address') || '';
-    return res(ctx.status(200), ctx.json(apiResponses?.[address]));
+const restHandlers = [
+  http.all('https://aha.rainbow.me/', ({ request }) => {
+    const url = new URL(request.url);
+    const address = url.searchParams.get('address') || '';
+    return HttpResponse.json(apiResponses?.[address], { status: 200 });
+  }),
+  http.options('*', () => {
+    return new Response(null, {
+      status: 200,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE',
+        'Access-Control-Allow-Headers': '*',
+      },
+    });
```

Comment:
Was this used to patch the `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource` error? If so, I think it was working as intended to warn us about an external domain fetch. Not used in prod obviously, but might catch supply-chain attacks

Reply:
Yes it was to patch that. what is our course of action if we see failures because of this? Is this something we WANT to get caught by unit tests? or something we should guard against elsewhere?

Reply:
okay i changed this to be more narrow and only allow through `rainbow.swaps` requests and even then, it still flags them with a `console.warn`